## The data in Table 5.1 samples of size 50 from a normal distribution, a skewed-right distribution, a skewed-left distribution, and a uniform distribution.
## The normal data are female height measurements, the skewed-right data are age at marriage for females, the skewed-left data are obituary data that give the age at death for females, and the uniform data are the amount of cola put into a 12 ounce container by a soft drinks machine.

In [248]:
import pandas as pd

In [249]:
df = pd.read_csv("/kaggle/input/table5-1/FA4 Data.csv")
df

Normal  Skewed-right  Skewed-left  Uniform
0       67            31          102     12.1
1       70            43           55     12.1
2       63            30           70     12.4
3       65            30           95     12.1
4       68            38           73     12.1
5       60            26           79     12.2
6       70            29           60     12.2
7       64            55           73     12.2
8       69            46           89     11.9
9       61            26           85     12.2
10      66            29           72     12.3
11      65            57           92     12.3
12      71            34           76     11.7
13      62            34           93     12.3
14      66            36           76     12.3
15      68            40           97     12.4
16      64            28           10     12.4
17      67            26           70     12.1
18      62            66           85     12.4
19      66            63           25     12.4
20      65            30           83     12.5
21      63            33           58     11.8
22      66            24           10     12.5
23      65            35           92     12.5
24      63            34           82     12.5
25      69            40           87     11.6
26      62            24          104     11.6
27      67            29           75     12.0
28      59            24           80     11.6
29      66            27           66     11.6
30      65            35           93     11.7
31      63            33           90     12.3
32      65            75           84     11.7
33      60            38           73     11.7
34      67            34           98     11.7
35      64            85           79     11.8
36      68            29           35     12.5
37      61            40           71     11.8
38      69            41           90     11.8
39      65            35           71     11.8
40      62            26           63     11.9
41      67            34           58     11.9
42      70            19           82     11.9
43      64            23           72     12.2
44      63            28           93     11.9
45      68            26           44     12.0
46      64            31           65     11.9
47      65            25           77     12.0
48      61            22           81     12.0
49      66            28           77     12.0

> ***Table 5.1***

## 1. Find the (a) first, (b) second, (c) third, and (d) fourth moments for each of the  sets of data (normal, skewed-right, skewed-left, uniform).

### (a) First moments

In [250]:
normal_first_moment = df['Normal'].mean()
print(f'First moment of the normal dataset: {normal_first_moment}')

First moment of the normal dataset: 65.12


In [251]:
skewed_right_first_moment = df['Skewed-right'].mean()
print(f'First moment of the skewed-right dataset: {skewed_right_first_moment}')

First moment of the skewed-right dataset: 35.48


In [252]:
skewed_left_first_moment = df['Skewed-left'].mean()
print(f'First moment of the skewed-left dataset: {skewed_left_first_moment}')

First moment of the skewed-left dataset: 74.2


In [253]:
uniform_first_moment = round(df['Uniform'].mean(), 3)
print(f'First moment of the uniform dataset: {uniform_first_moment}')

First moment of the uniform dataset: 12.056


### (b) Second moments

In [254]:
normal_squared = df['Normal']**2
normal_second_moment = normal_squared.mean()
print(f'Second moment of the normal dataset: {normal_second_moment}')

Second moment of the normal dataset: 4248.92


In [255]:
skewed_right_squared = df['Skewed-right']**2
skewed_right_second_moment = skewed_right_squared.mean()
print(f'Second moment of the skewed-right dataset: {skewed_right_second_moment}')

Second moment of the skewed-right dataset: 1437.72


In [256]:
skewed_left_squared = df['Skewed-left']**2
skewed_left_second_moment = skewed_left_squared.mean()
print(f'Second moment of the skewed-left dataset: {skewed_left_second_moment}')

Second moment of the skewed-left dataset: 5925.4


In [257]:
uniform_squared = df['Uniform']**2
uniform_second_moment = uniform_squared.mean()
print(f'Second moment of the uniform dataset: {uniform_second_moment}')

Second moment of the uniform dataset: 145.426


### (c) Third moments

In [258]:
normal_cube = df['Normal']**3
normal_third_moment = normal_cube.mean()
print(f'Third moment of the normal dataset: {normal_third_moment}')

Third moment of the normal dataset: 277770.92


In [259]:
skewed_right_cube = df['Skewed-right']**3
skewed_right_third_moment = skewed_right_cube.mean()
print(f'Third moment of the skewed-right dataset: {skewed_right_third_moment}')

Third moment of the skewed-right dataset: 68292.44


In [260]:
skewed_left_cube = df['Skewed-left']**3
skewed_left_third_moment = skewed_left_cube.mean()
print(f'Third moment of the skewed-left dataset: {skewed_left_third_moment}')

Third moment of the skewed-left dataset: 489458.8


In [261]:
uniform_cube = df['Uniform']**3
uniform_third_moment = uniform_cube.mean()
print(f'Third moment of the uniform dataset: {uniform_third_moment}')

Third moment of the uniform dataset: 1755.15776


### (c) Fourth moments

In [262]:
normal_quad = df['Normal']**4
normal_fourth_moment = normal_quad.mean()
print(f'Fourth moment of the normal dataset: {normal_fourth_moment}')

Fourth moment of the normal dataset: 18194173.64


In [263]:
skewed_right_quad = df['Skewed-right']**4
skewed_right_fourth_moment = skewed_right_quad.mean()
print(f'Fourth moment of the skewed-right dataset: {skewed_right_fourth_moment}')

Fourth moment of the skewed-right dataset: 3797594.04


In [264]:
skewed_left_quad = df['Skewed-left']**4
skewed_left_fourth_moment = skewed_left_quad.mean()
print(f'Fourth moment of the skewed-left dataset: {skewed_left_fourth_moment}')

Fourth moment of the skewed-left dataset: 41396161.48


In [265]:
uniform_quad = df['Uniform']**4
uniform_fourth_moment = uniform_quad.mean()
print(f'Fourth moment of the uniform dataset: {uniform_fourth_moment}')

Fourth moment of the uniform dataset: 21194.5933


## 2. Find the (a) first, (b) second, (c) third, and (d ) fourth moments about the mean for each of the sets of data (normal, skewed-right, skewed-left, uniform).

### (a) First moments about the mean

In [266]:
normal_minus_mean = df['Normal'] - normal_first_moment
normal_first_moment_mean = round(normal_minus_mean.mean(), 2)
print(f'First moment about the mean of the normal dataset: {normal_first_moment_mean}')

First moment about the mean of the normal dataset: -0.0


In [267]:
skewed_right_minus_mean = df['Skewed-right'] - skewed_right_first_moment
skewed_right_first_moment_mean = round(skewed_right_minus_mean.mean(), 2)
print(f'First moment about the mean of the skewed-right dataset: {skewed_right_first_moment_mean}')

First moment about the mean of the skewed-right dataset: 0.0


In [268]:
skewed_left_minus_mean = df['Skewed-left'] - skewed_left_first_moment
skewed_left_first_moment_mean = round(skewed_left_minus_mean.mean(), 2)
print(f'First moment about the mean of the skewed-left dataset: {skewed_left_first_moment_mean}')

First moment about the mean of the skewed-left dataset: -0.0


In [269]:
uniform_minus_mean = df['Uniform'] - uniform_first_moment
uniform_first_moment_mean = round(uniform_minus_mean.mean(), 2)
print(f'First moment about the mean of the uniform dataset: {uniform_first_moment_mean}')

First moment about the mean of the uniform dataset: 0.0


### (b) Second moments about the mean

In [270]:
normal_var = normal_minus_mean ** 2
normal_second_moment_mean = round(normal_var.mean(), 2)
print(f'Second moment about the mean of the normal dataset: {normal_second_moment_mean}')

Second moment about the mean of the normal dataset: 8.31


In [271]:
skewed_right_var = skewed_right_minus_mean ** 2
skewed_right_second_moment_mean = round(skewed_right_var.mean(), 2)
print(f'Second moment about the mean of the skewed-right dataset: {skewed_right_second_moment_mean}')

Second moment about the mean of the skewed-right dataset: 178.89


In [272]:
skewed_left_var = skewed_left_minus_mean ** 2
skewed_left_second_moment_mean = round(skewed_left_var.mean(), 2)
print(f'Second moment about the mean of the skewed-left dataset: {skewed_left_second_moment_mean}')

Second moment about the mean of the skewed-left dataset: 419.76


In [273]:
uniform_var = uniform_minus_mean ** 2
uniform_second_moment_mean = round(uniform_var.mean(), 2)
print(f'Second moment about the mean of the uniform dataset: {uniform_second_moment_mean}')

Second moment about the mean of the uniform dataset: 0.08


### (c) Third moments about the mean

In [274]:
normal_m_mean_cube = normal_minus_mean ** 3
normal_third_moment_mean = round(normal_m_mean_cube.mean(), 2)
print(f'Third moment about the mean of the normal dataset: {normal_third_moment_mean}')

Third moment about the mean of the normal dataset: -0.47


In [275]:
skewed_right_m_mean_cube = skewed_right_minus_mean ** 3
skewed_right_third_moment_mean = round(skewed_right_m_mean_cube.mean(), 2)
print(f'Third moment about the mean of the skewed-right dataset: {skewed_right_third_moment_mean}')

Third moment about the mean of the skewed-right dataset: 4588.13


In [276]:
skewed_left_m_mean_cube = skewed_left_minus_mean ** 3
skewed_left_third_moment_mean = round(skewed_left_m_mean_cube.mean(), 5)
print(f'Third moment about the mean of the skewed-left dataset: {skewed_left_third_moment_mean}')

Third moment about the mean of the skewed-left dataset: -12498.264


In [277]:
uniform_m_mean_cube = uniform_minus_mean ** 3
uniform_third_moment_mean = round(uniform_m_mean_cube.mean(), 2)
print(f'Third moment about the mean of the uniform dataset: {uniform_third_moment_mean}')

Third moment about the mean of the uniform dataset: 0.0


### (d) Fourth moments about the mean

In [278]:
normal_m_mean_quad = normal_minus_mean ** 4
normal_fourth_moment_mean = round(normal_m_mean_quad.mean(), 2)
print(f'Fourth moment about the mean of the normal dataset: {normal_fourth_moment_mean}')

Fourth moment about the mean of the normal dataset: 160.95


In [279]:
skewed_right_m_mean_quad = skewed_right_minus_mean ** 4
skewed_right_fourth_moment_mean = round(skewed_right_m_mean_quad.mean(), 2)
print(f'Fourth moment about the mean of the skewed-right dataset: {skewed_right_fourth_moment_mean}')

Fourth moment about the mean of the skewed-right dataset: 210642.88


In [280]:
skewed_left_m_mean_quad = skewed_left_minus_mean ** 4
skewed_left_fourth_moment_mean = round(skewed_left_m_mean_quad.mean(), 2)
print(f'Fourth moment about the mean of the skewed-left dataset: {skewed_left_fourth_moment_mean}')

Fourth moment about the mean of the skewed-left dataset: 927289.75


In [281]:
uniform_m_mean_quad = uniform_minus_mean ** 4
uniform_fourth_moment_mean = round(uniform_m_mean_quad.mean(), 2)
print(f'Fourth moment about the mean of the uniform dataset: {uniform_fourth_moment_mean}')

Fourth moment about the mean of the uniform dataset: 0.01


## 3. Find the (a) first, (b) second, (c) third, and (d ) fourth moments about the number 75 for the set of female height measurements.

### (a) First moment about the number 75

In [282]:
normal_minus_75 = df['Normal'] - 75
normal_first_moment_75 = round(normal_minus_75.mean(), 2)
print(f'''First moment about the number 75 of the set of female height measurements:
{normal_first_moment_75}''')

First moment about the number 75 of the set of female height measurements:
-9.88


### (b) Second moment about the number 75

In [283]:
normal_75_squared = normal_minus_75 ** 2
normal_second_moment_75 = round(normal_75_squared.mean(), 2)
print(f'''Second moment about the number 75 of the set of female height measurements:
{normal_second_moment_75}''')

Second moment about the number 75 of the set of female height measurements:
105.92


### (c) Third moment about the number 75

In [284]:
normal_75_cube = normal_minus_75 ** 3
normal_third_moment_75 = round(normal_75_cube.mean(), 2)
print(f'''Third moment about the number 75 of the set of female height measurements:
{normal_third_moment_75}''')

Third moment about the number 75 of the set of female height measurements:
-1211.08


### (d) Fourth moment about the number 75

In [285]:
normal_75_quad = normal_minus_75 ** 4
normal_fourth_moment_75 = round(normal_75_quad.mean(), 2)
print(f'''Fourth moment about the number 75 of the set of female height measurements:
{normal_fourth_moment_75}''')

Fourth moment about the number 75 of the set of female height measurements:
14572.64


## 4. Using the results of items 2 and 3 for the set of female height measurements, verify the relations between the moments (a) $m_2 = m'_2 - m'^{2}_1$, (b) $m_3 = m'_3 - 3m'_1m'_2 + 2m'^{3}_1$, and (c) $m_4 = m'_4 - 4m'_1m'_3 + 6m'^{2}_1m'_2 - 3m'^{4}_1$

In [286]:
# Moments about the number 75
m_prime_1 = round(normal_first_moment_75, 2)
m_prime_2 = round(normal_second_moment_75, 2)
m_prime_3 = round(normal_third_moment_75, 2)
m_prime_4 = round(normal_fourth_moment_75, 2)

# Moments about the mean
m_2 = round(normal_second_moment_mean, 2)
m_3 = round(normal_third_moment_mean, 2)
m_4 = round(normal_fourth_moment_mean, 2)

### (a) Verifying $m_2 = m'_2 - m'^{2}_1$

In [287]:
if m_2 == round(m_prime_2 - round((m_prime_1 ** 2), 2), 2):
    print("They are equal")
else:
    print("They are not equal.")

They are equal


$$
m_2 = m'_2 - m'^{2}_1$ \\
m_2 = 105.92 - (-9.88)^2 \\
m_2 = 8.31 \ \checkmark
$$
**This is in agreement with item 2 $\checkmark$**

### (b) Verifying $m_3 = m'_3 - 3m'_1m'_2 + 2m'^{3}_1$

In [288]:
if m_3 == round(m_prime_3 - round((3*m_prime_1*m_prime_2), 2) + round(2*round(m_prime_1**3, 2), 2), 2):
    print("They are equal")
else:
    print("They are not equal.")

They are equal


$$
m_3 = m'_3 - 3m'_1m'_2 + 2m'^{3}_1 \\
m_3 = -1211.08 - 3(-9.88)(105.92) + 2(-9.88)^3 \\
m_3 = -1211.08 + 3139.47 + 2(-964.43) \\
m_3 = -1211.08 + 3139.47 - 1928.86 \ \checkmark
$$
**This is in agreement with item 2 $\checkmark$**

### (c) Verifying $m_4 = m'_4 - 4m'_1m'_3 + 6m'^{2}_1m'_2 - 3m'^{4}_1$

In [289]:
if m_4 == round(m_prime_4 - (4*m_prime_1*m_prime_3) + (6*(m_prime_1**2)*m_prime_2) - (3*(m_prime_1**4)), 2) :
    print("They are equal")
else:
    print("They are not equal.")

They are equal


$$
m_4 = m'_4 - 4m'_1m'_3 + 6m'^{2}_1m'_2 - 3m'^{4}_1 \\
m_4 = 14572.64 - 4(-9.88)(-1211.08) + 6(-9.88)^2(105.92) - 3(-9.88)^4 \\
m_4 = 14572.64 - 47861.8816 + 62035.9 - 28 585.71 \\
m_4 = 160.95 \ \checkmark
$$
**This is in agreement with item 2 $\checkmark$**